R/areaplot.R

```diff
@@ -141,9 +142,17 @@ AreaPlot <- function(
         datas <- split(data, data[[split_by]])
         # keep the order of levels
         datas <- datas[levels(data[[split_by]])]
+        palette <- check_palette(palette, names(datas))
+        palcolor <- check_palcolor(palcolor, names(datas))
     } else {
         datas <- list(data)
+        palette <- list(palette)
         names(datas) <- "..."
+        names(palette) <- "..."
+        if (!is.null(palcolor)) {
+	        palcolor <- list(palcolor)
+        	palcolor <- check_palcolor(palcolor, "...")
+        }
```

Comment:
This together can be implemented in `check_palcolor()`. If it is `NULL`, just return `NULL`.